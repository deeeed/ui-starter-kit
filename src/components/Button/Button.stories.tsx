import type { Meta, StoryObj } from '@storybook/react-native';
import { Button } from './Button';

const ButtonMeta: Meta<typeof Button> = {
  component: Button,
  argTypes: {
    onPress: { action: 'pressed the button' },
  },
  args: {
    children: 'Hello world',
  },
};

export default ButtonMeta;
type Story = StoryObj<typeof Button>;

export const Basic: Story = {
  name: 'Test Basic',
};

export const AnotherExample: Story = {
  name: 'Contained',
  args: {
    mode: 'contained',
  },
};
