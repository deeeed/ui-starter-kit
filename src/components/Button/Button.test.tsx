import * as React from 'react';

import { render } from '@testing-library/react-native';
import { Button } from './Button';

it('renders text button by default', () => {
  const tree = render(<Button>Text Button</Button>).toJSON();

  expect(tree).toMatchSnapshot();
});
