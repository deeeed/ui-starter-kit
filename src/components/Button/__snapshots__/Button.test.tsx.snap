// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders text button by default 1`] = `
<View
  collapsable={false}
  style={
    Object {
      "backgroundColor": "transparent",
      "borderRadius": 20,
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 0,
        "width": 0,
      },
      "shadowOpacity": 0,
      "shadowRadius": 0,
    }
  }
  testID="button-container-outer-layer"
>
  <View
    collapsable={false}
    style={
      Object {
        "backgroundColor": "transparent",
        "borderColor": "transparent",
        "borderRadius": 20,
        "borderStyle": "solid",
        "borderWidth": 0,
        "flex": undefined,
        "minWidth": 64,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 0,
        "shadowRadius": 0,
      }
    }
    testID="button-container"
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        Object {
          "busy": undefined,
          "checked": undefined,
          "disabled": true,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        Object {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Object {
            "borderRadius": 20,
          },
        ]
      }
      testID="button"
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
            },
            undefined,
          ]
        }
      >
        <Text
          numberOfLines={1}
          selectable={false}
          style={
            Array [
              Object {
                "textAlign": "left",
              },
              Object {
                "color": "rgba(28, 27, 31, 1)",
                "writingDirection": "ltr",
              },
              Array [
                Object {
                  "fontFamily": "System",
                  "fontSize": 14,
                  "fontWeight": "500",
                  "letterSpacing": 0.1,
                  "lineHeight": 20,
                },
                Array [
                  Object {
                    "marginHorizontal": 16,
                    "marginVertical": 9,
                    "textAlign": "center",
                  },
                  false,
                  Object {
                    "marginHorizontal": 12,
                  },
                  undefined,
                  false,
                  Object {
                    "color": "rgba(103, 80, 164, 1)",
                    "fontFamily": "System",
                    "fontSize": 14,
                    "fontWeight": "500",
                    "letterSpacing": 0.1,
                    "lineHeight": 20,
                  },
                  undefined,
                ],
              ],
            ]
          }
          testID="button-text"
        >
          Text Button
        </Text>
      </View>
    </View>
  </View>
</View>
`;
